microbench/src/main/java/io/netty/microbenchmark/common/IsValidIpV6Benchmark.java

```diff
@@ -0,0 +1,367 @@
+            c = ip.charAt(i);
+            switch (c) {
+            case ':':
+                if (colons > 7) {
+                    return false;
+                }
+                if (ip.charAt(i - 1) == ':') {
+                    if (compressBegin >= 0) {
+                        return false;
+                    }
+                    compressBegin = i - 1;
+                } else {
+                    wordLen = 0;
+                }
+                colons++;
+                break;
+            case '.':
+                // case for the last 32-bits represented as IPv4 x:x:x:x:x:x:d.d.d.d
+
+                // check a normal case (6 single colons)
+                if (compressBegin < 0 && colons != 6 ||
+                    // a special case ::1:2:3:4:5:d.d.d.d allows 7 colons with an
+                    // IPv4 ending, otherwise 7 :'s is bad
+                    (colons == 7 && compressBegin >= start || colons > 7)) {
+                    return false;
+                }
+
+                // Verify this address is of the correct structure to contain an IPv4 address.
+                // It must be IPv4-Mapped or IPv4-Compatible
+                // (see https://tools.ietf.org/html/rfc4291#section-2.5.5).
+                int ipv4Start = i - wordLen;
+                int j = ipv4Start - 2; // index of character before the previous ':'.
+                if (isValidIPv4MappedChar(ip.charAt(j))) {
+                    if (!isValidIPv4MappedChar(ip.charAt(j - 1)) ||
+                        !isValidIPv4MappedChar(ip.charAt(j - 2)) ||
+                        !isValidIPv4MappedChar(ip.charAt(j - 3))) {
+                        return false;
+                    }
+                    j -= 5;
+                }
+
+                for (; j >= start; --j) {
+                    char tmpChar = ip.charAt(j);
+                    if (tmpChar != '0' && tmpChar != ':') {
+                        return false;
+                    }
+                }
+
+                int ipv4End = ip.indexOf('%', ipv4Start + 7);
+                if (ipv4End < 0) {
+                    ipv4End = end;
+                }
+                return isValidIpV4Address(ip, ipv4Start, ipv4End);
+            case '%':
+                // Strip the interface name/index after the percent sign
+                end = i;
+                break loop;
+            default:
+                if (wordLen > 3 || !isValidHexChar(c)) {
+                    return false;
+                }
+                wordLen++;
+            }
+        }
+
+        // normal case without compression
+        if (compressBegin < 0) {
+            return colons == 7 && wordLen > 0;
+        }
+
+        return compressBegin + 2 == end ||
+               // 8 colons is valid only if compression in start or end
+               wordLen > 0 && (colons < 8 || compressBegin <= start);
+    }
+
+    // Tests
+
+//    @Benchmark
+//    public boolean isValidIpV6AddressOld() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3ACommentedOutCodeLine)